src/Views/BehaviorView.vala

```diff
@@ -145,25 +135,24 @@ public class Power.MainView : Gtk.Grid {
             permission.bind_property ("allowed", lid_dock_label, "sensitive", GLib.BindingFlags.SYNC_CREATE);
             permission.bind_property ("allowed", lock_image, "visible", GLib.BindingFlags.SYNC_CREATE | GLib.BindingFlags.INVERT_BOOLEAN);
             permission.bind_property ("allowed", lock_image2, "visible", GLib.BindingFlags.SYNC_CREATE | GLib.BindingFlags.INVERT_BOOLEAN);
-            permission.bind_property ("allowed", permission_infobar, "revealed", GLib.BindingFlags.SYNC_CREATE | GLib.BindingFlags.INVERT_BOOLEAN);
         }
 
         var screen_timeout_label = new Gtk.Label (_("Turn off display when inactive for:"));
         screen_timeout_label.halign = Gtk.Align.END;
         screen_timeout_label.xalign = 1;
 
-        var screen_timeout = new TimeoutComboBox (new GLib.Settings ("org.gnome.desktop.session"), "idle-delay");
-
+        var screen_timeout = new TimeoutComboBox (elementary_dpms_settings, "standby-time");
+        screen_timeout.changed.connect (run_dpms_helper);
```

Comment:
This looks like a behavior change or at least a bug fix; if you want to break things down further, this could probably be proposed as its own bitesize PR.

Reply:
Done!